src/WebJobs.Script.WebHost/Models/RunFromPackageContext.cs

```diff
@@ -31,14 +26,19 @@ public RunFromPackageContext(string envVarName, string url, long? packageContent
 
         public bool IsScmRunFromPackage()
         {
-            return string.Equals(EnvironmentVariableName, EnvironmentSettingNames.ScmRunFromPackage,
-                        StringComparison.OrdinalIgnoreCase);
+            return string.Equals(EnvironmentVariableName, EnvironmentSettingNames.ScmRunFromPackage, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public bool IsRunFromPackage(ScriptApplicationHostOptions options, ILogger logger)
+        {
+            return (IsScmRunFromPackage() && ScmRunFromPackageBlobExists(options, logger)) || (!IsScmRunFromPackage() && !string.IsNullOrEmpty(Url) && Url != "1");
```

Comment:
Can't we skip the IsScmRunFromPackage checks here. When you set the new options member you already check for settings and blob existence. Seems like this logic can be simplified.

Reply:
I think you're right, they're duplicate checks. cc @balag0 just to confirm there's no issues with this approach.

The new method can just be:
```
public bool IsRunFromPackage(ScriptApplicationHostOptions options, ILogger logger)
{
            var blobExists = options.ScmRunFromPackageBlobExists;
            logger.LogDebug($"{EnvironmentSettingNames.ScmRunFromPackage} points to an existing blob: {blobExists}");
}
```

Reply:
Ah, as I typed this comment I realized this is not sufficient. `RunFromPackageContext` includes other types of run from package, like from these app settings:
```
public const string AzureWebsiteZipDeployment = "WEBSITE_USE_ZIP";
public const string AzureWebsiteAltZipDeployment = "WEBSITE_RUN_FROM_ZIP";
public const string AzureWebsiteRunFromPackage = "WEBSITE_RUN_FROM_PACKAGE";
```

Which is why we have this part of the check: `|| (!IsScmRunFromPackage() && !string.IsNullOrEmpty(Url) && Url != "1");`

Reply:
I was just wondering why your check couldn't be like:

```csharp
return options.IsScmRunFromPackage || (!IsScmRunFromPackage() && !string.IsNullOrEmpty(Url) && Url != "1");
```

Anyhow, if Bala is happy with the logic, I'm ok I guess

Reply:
Yes. this should also work and is better. I misunderstood the comment earlier and I just wanted to make sure we retain the 
(!IsScmRunFromPackage() && !string.IsNullOrEmpty(Url) && Url != "1") part of it :)